pydicom/filereader.py

```diff
@@ -230,6 +233,15 @@ def data_element_generator(fp,
     element_struct_unpack = element_struct.unpack
     defer_size = size_in_bytes(defer_size)
 
+    tag_set = None
+    if specific_tags is not None:
+        tag_set = set()
```

Comment:
Assessing the empty set is faster than using None
```py
u = None
v = set()

In [14]: %timeit u == None
The slowest run took 38.24 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 38.3 ns per loop

In [15]: %timeit not v
The slowest run took 62.51 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 32.5 ns per loop
```
And I find explicit union more readable

```py
if specific_tags and tag not in specific_tags.union(extra_tags):
```